Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -306,13 +327,16 @@ extension MongoCollection {
      *
      * - Returns: A `MongoCursor` over the index names.
      *
-     * - Throws: A `userError.invalidArgumentError` if the options passed are an invalid combination.
+     * - Throws: `UserError.logicError` if the provided session is inactive.
      */
-    public func listIndexes() throws -> MongoCursor<Document> {
-        guard let cursor = mongoc_collection_find_indexes_with_opts(self._collection, nil) else {
+    public func listIndexes(session: ClientSession? = nil) throws -> MongoCursor<Document> {
+        // need this cast to infer a generic type
+        let opts = try encodeOptions(options: nil as DropIndexOptions?, session: session, using: self.encoder)
```

Comment:
lol TIL you can cast nil... makes sense but kind of silly. did you consider adding a default value of nil for the options parameter to `encodeOptions` instead?